tests/test_mpi.py

```diff
@@ -1411,8 +1446,8 @@ def test_adjoint_codegen(self, shape, kernel, space_order, nbpml, save):
         op_adj = solver.op_adj()
         adj_calls = FindNodes(Call).visit(op_adj)
 
-        assert len(fwd_calls) == 2
-        assert len(adj_calls) == 2
+        assert len(fwd_calls) == 1
+        assert len(adj_calls) == 1
```

Comment:
Why did this change again? Better halo exchange?

Reply:
yes

One less time-invariant halo exchange